implementations/rust/ockam/ockam/src/entity/profile/profile_change_type/create_key.rs

```diff
@@ -0,0 +1,100 @@
+use crate::entity::profile::event_identifier::EventIdentifier;
+use crate::entity::profile::profile_change::{ProfileChange, ProfileKeyPurpose, ProfileKeyType};
+use crate::entity::profile::profile_change_type::ProfileChangeType;
+use crate::entity::profile::proof::{ProfileChangeProof, Signature, SignatureType};
+use crate::entity::{Changes, Profile, ProfileChangeEvent, ProfileEventAttributes};
+use crate::OckamError;
+use ockam_vault_core::types::{SecretAttributes, SecretPersistence, SecretType};
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct CreateKeyChange {
+    label: String,
+    key_type: ProfileKeyType,
+    key_purpose: ProfileKeyPurpose,
+    public_key: Vec<u8>,
```

Comment:
Do we want to combine key_type and public key into an enum that wraps the type and value together?

Reply:
@mikelodder7 yes, that's something we needed to do while working on vault. Still thinks this should be implemented inside vault crate

Reply:
agreed.

Reply:
@mikelodder7 how about throwing kid inside that struct as well?